config/config.go

```diff
@@ -16,15 +16,56 @@ const OPT_OUTPUT_FOLDER = "output-folder"
 const OPT_NON_INTERACTIVE = "non-interactive"
 const OPT_VAR = "var"
 const OPT_VAR_FILE = "var-file"
+const OPT_MISSING_KEY_ACTION = "missing-key-action"
 
 // The command-line options for the boilerplate app
 type BoilerplateOptions struct {
-	TemplateFolder 	string
-	OutputFolder 	string
-	NonInteractive	bool
-	Vars		map[string]string
+	TemplateFolder 	 string
+	OutputFolder 	 string
+	NonInteractive	 bool
+	Vars		 map[string]string
+	OnMissingKey     MissingKeyAction
 }
 
+// This type is an enum that represents what we can do when a template looks up a missing key. This typically happens
+// when there is a typo in the variable name in a template.
+type MissingKeyAction int
+
+func (action MissingKeyAction) String() string {
+	return missingKeyNames[int(action)]
+}
+
+// Convert the given string to a MissingKeyAction enum, or return an error if this is not a valid value for the
+// MissingKeyAction enum
+func ParseMissingKeyAction(keyName string) (MissingKeyAction, error) {
+	for i, missingKeyName := range missingKeyNames {
+		if missingKeyName == keyName {
+			return MissingKeyAction(i), nil
+		}
+	}
+	return MissingKeyAction(-1), errors.WithStackTrace(InvalidMissingKeyAction(keyName))
+}
+
+// The names of the missing keys. Go doesn't have enums, so we have to roll our own based on this example:
+// https://gist.github.com/skarllot/102a5e5ea73861ff5afe
+var missingKeyNames = []string{}
```

Comment:
Oh, I'm guessing it's because you want to convert these to strings. Could you just define make `MissingKeyAction` of type `string` and initialize it to a sane string literal?